### ISSUE TITLE
The project name of the "TheScriptRunner"

### ISSUE THREAD

**Issue by piqus** [TheScriptRunner/desktop-app#1]
Current name is something like codename for whole project. In my humble opinnion we should change it to be more marketing-friendly purposes.

I am opened for any suggestions, please type then below :)

**Comment by ovirendo** [comment]
- plugger - not bad, but it is difficult to popularize it (sounds like application for Nest platform)
- windows-tweak-tool - sounds like crappy "speed boost your pc" advert, plus we are going to be cross-OS.

**Comment by piqus** [comment]
- plugger - not bad, but it is difficult to popularize it (sounds like application for Nest platform)
- windows-tweak-tool - sounds like crappy "speed boost your pc" advert, plus we are going to be cross-OS.

**Comment by 10se1ucgo** [comment]
I think we should just get straight to the point in what it does.

i.e. Names like Windows Tweak Tool doesn't really explain what it tweaks, but things like Debloat Windows and Disable Windows Tracking sort of explains exactly what the program is meant to do.

**Comment by tomchappelow** [comment]
And, is it being aimed at sysadmins, or home users? That will have a big impact

**Comment by 10se1ucgo** [comment]
sysadmins usually should have less to fear when it comes to privacy, as people really shouldn't be doing private stuff in workplaces and school, but that depends what the sysadmin is administrating for.

**Comment by tomchappelow** [comment]
In 2015, most enterprise admins have an appreciation of privacy--who wants to give anybody access to anything in your environment (when it's forced on you)?

I'm favouring, like you, something straight to the point.

Users and admins alike are busy people, and both will be welcoming of a descriptive name. But one that still catches attention; perhaps something like 'Privacy Tool' with a tag line of Disable Windows Tracking?

**Comment by 10se1ucgo** [comment]
Hmm, considering a sysadmin should have some sort of knowledge with command-line interfaces, I think they should be alright with PowerShell scripts.

For the end-user, I think creating a pretty (this is important, odd / sketchy looking interfaces can drive people away), native (as in, fits the theme of the OS, so Metro for Windows 8/10, Aero for Vista/7, Luna for XP, etc.) GUIs will be good. We can accomplish this with wxWidgets, or in my case, the bindings for python (wxPython).

**Comment by tomchappelow** [comment]
I agree. But we all know admins who prefer to run a tool--if we're going for a wide uptake, we need to cater for the masses. Just look at IIS Crypto's success with protocol hardening, admins love it. 

I think we agree on this, though, given the look of your tool ;)

**Comment by IMMORTAL-DEATH** [comment]
I was reading the first few comments about names what about Isolation or stealth patcher ?

**Comment by HarshaThota** [comment]
I don't know if naming it as Privacy Tool, Disable Windows Tracking, etc. will work. As I understand it, the scope of the app is to run any script, at least based on what [Plugger](https://github.com/Plugger-io/plugger) was aiming to build. If that's the case, I don't think you can name it as anything specific to Windows tracking/privacy related changes. In fact, Script Runner actually seems appropriate in that case.

**Comment by 10se1ucgo** [comment]
For sure, I'm just suggesting something that gets the point across.

**Comment by sehams** [comment]
@10se1ucgo using wxWidgets is neat, though we should create something more overseeable as this many checkboxes can be quite confusing, especially for a novize or someone not that much affiliated with PCs in general.

For a name I would suggest something like *Scribere* or *Scriptum*, both latin words from which the english word script derives.
Nevertheless @HarshaThota is right, we should look for a name that doesn't make people think it's designed to do one (or several tasks) and nothing more. Also, we should make sure that the name is catchy and easy to pronounce, even for foreigners. We could also demenstrote our openness in both the code and the platform itself by adding the popular *Open* in front of it.
So something like OpenScriptum would work for me. We could make it even more distinguished for search engines and in general by replacing the c with a k, making it **OpenSkriptum**

**Comment by piqus** [comment]
Yay! You're great, I really appreciate helping me to find a name for the project, I like them all :+1: 

I must agree wth many of you in the same moment :smile_cat: The name must name what it is?, and what it do? (same argument as @10se1ucgo mentioned). But it needs to tell us that is not only windows-compatible or tool is made for privacy things. 

Well, but probably I am doing it wrong to connect to separate things into one application. We did (except me) a great job connected with win-10 privacy "features", hardening the firewalls and debloating the OS. But hey, it is a our feature-requirement to run it as a script from the app, right?

**Comment by sehams** [comment]
I think we should concentrate more on the catchyness of the name. Sure it's good to have a to-the-point name, but a tool capable of all the thing we imagine would have quite a long name, wouldn't it ;)
It has to be a name that sticks and that people do not associate with anything else. If it's a good tool it will find it's popularity besides the name anyway, just think about the popular *CCleaner*, *TuneUp Utilities*, and so on. These arent great names either.

**Comment by HarshaThota** [comment]
This is building on some of the comments from nathan-alden/windows-10-tracking#6 but is it really necessary for this tool to be x-platform?

I would expect that anyone using Linux and is running scripts, would be comfortable enough with the command line that they wouldn't necessarily need a GUI tool. I think Windows is an exception in that due to its wide-spread usage, you have people that don't know command line but are in a position to administer and maintain their machines. 

I think it still makes sense using Electron just due to the ease of development but I am wondering if it's really necessary to go x-platform.

**Comment by sehams** [comment]
@HarshaThota The way we plan to program it wont be much work to put in for making it work on any OS. Actually the sript only has to be marked for the OS you're currently using.
Even though you're right and this is more or less a Windows problem it's still gold for admins who habe to do a lot of the same tasks over and over again. Even on Linux. Also there are a lot of affiliated Linux users who STILL do not have all the neccesary skills to do everything on their own.

**Comment by sehams** [comment]
Also we won't be able to do much about the cross-compatibility anyways. If its open source (and greatly coded) people will be able compile it for their platform and start uploading scripts :D

**Comment by 10se1ucgo** [comment]
Mockup written in Python with wxPython, based off @sehams's design.

![image](http://i.imgur.com/R3ktnzl.png)
![gif](http://i.imgur.com/Qkj7Yzj.gif)

**Comment by piqus** [comment]
@HarshaThota I will add to @sehams one pros for open source argument:

- Electron is x-platform OOTB
- Node is x-platform OOTB
- Python is x-platform OOTB (well, why not use it for development purposes:P)

[OOTB]: Out of the box.

@10se1ucgo It is nice too see this mockup IR.
